File: programmers/스킬체크테스트/level2/1번째/1.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

// nums_len은 배열 nums의 길이입니다.
int solution(int nums[], size_t nums_len) {
    int answer = 0;
    int i = 0, j = 0, k = 0, n = 0, m = 0;
    int com[3] = {0,};
    int sum = 0;
    int check = 0;

    for(i = 0; i < nums_len - 2; i++) {
        com[0] = nums[i];
        sum += com[0];
        for(j = i+1; j < nums_len - 1; j++) {
            com[1] = nums[j];
            sum += com[1];
            for( k = j+1; k < nums_len; k++) {
                com[2] = nums[k];
                sum += com[2];
                
                //소수 계산
                for(m = 2; m < sum; m++) {
                    //printf("%d / %d = %d\n", sum, m, sum%m);
                    if(sum % m == 0) {
                        check = 1;
                        break;
                    }
                }
                if (check == 0) answer++;
                check = 0;
                sum -= com[2];
            }
            sum -= com[1];
        }
        sum -= com[0];
    } 


    return answer;
}


int main() {
    int nums[] = {1, 2, 3, 4};
    printf("%d", solution(nums, 4));

    return 0;
}

File: 코딩테스트/2021_네이버웹툰_개발챌린지/1.py
def solution(lottery):
    answer = -1

    dic = {}
    for i in range(1, 1001):
        dic[i] = [False, 0]

    count = []
    # i 아이디, j 당첨여부
    for i, j in lottery:
        if dic[i][0] == False:
            dic[i][1] += 1

            if j == 1:
                dic[i][0] = True
                count.append(dic[i][1])
        
    
    if not count:
        answer = 0
    else:
        answer = sum(count)//len(count)

    return answer


print(solution([[1,0],[35,0],[1,0],[100,1],[35,1],[100,1],[35,0],[1,1],[1,1]]))
print(solution([[1,0],[2,0],[3,0],[1,0],[2,0],[1,0],[1,1],[2,0],[2,0],[2,1],[1,0],[1,1],[3,0],[3,0],[1,1]]))
print(solution([[1,0],[2,0],[3,0]]))